client/header/activity-panel/panel.js

```diff
@@ -31,6 +31,12 @@ export const Panel = ( {
 		}
 	};
 
+	const finishTransition = ( e ) => {
+		if ( e && e.propertyName === 'transform' ) {
+			clearPanel();
```

Comment:
Only trigger this when the propertyName is `transform`, it is triggered for several others as well.